FirebaseDatabase/Sources/Core/FSyncTree.m

```diff
@@ -575,6 +620,23 @@ - (FCacheNode *)serverCacheForQuery:(FQuerySpec *)query {
     return serverCache;
 }
 
+- (void)unregisterQuery:(FQuerySpec *)query {
+    FPath *path = query.path;
+    FSyncPoint *maybeSyncPoint = [self.syncPointTree valueAtPath:path];
+
+    if (maybeSyncPoint &&
+        ([query isDefault] || [maybeSyncPoint viewExistsForQuery:query])) {
+        BOOL removed = [maybeSyncPoint unregisterQuery:query];
+        if ([maybeSyncPoint isEmpty]) {
+            self.syncPointTree = [self.syncPointTree removeValueAtPath:path];
+        }
+        if (removed) {
+            [self removeTags:@[ query ]];
+            [self.persistenceManager setQueryInactive:query];
```

Comment:
Note: only call setQueryInactive when we just removed the last query in the view. This is consistent with how `removeEventRegistration` works.